### DISCUSSION TITLE
Modify screen width

### DISCUSSION THREAD

**Discussion by Ultramxn** [olikraus/u8g2#2140] [Q&A] [answered]
Hello,
I have a ST7657 12232 LCD cannot display anything with 12232 constructor in u8g2 but display well with OS12864 constructor except skipping six pixels on the left of the screen, that is, missing first character in the test. May I ask can we override screen width of a constructor by passing parameter?

**Comment by olikraus** [comment, marked as answer]
What are the exact names of the constructors?

**Reply by Ultramxn** [reply]
![20230323_034404](https://user-images.githubusercontent.com/128638756/227503235-a5504e1a-cbe7-44b8-bab3-a534ed17cc1e.jpg)
![20230323_034151](https://user-images.githubusercontent.com/128638756/227503254-1a2ef1c4-b844-427f-8d44-adad3d34cad1.jpg)
Graphics test results are uploaded for your convenience.

Many thanks.

**Comment by Ultramxn** [comment]
Thanks for your prompt reply !

The constructors are:
U8G2_ST7567_122X32_F_4W_SW_SPI // which displayed nothing.
and
U8G2_ST7567_OS12864_F_4W_SW_SPI // display test screen but missing left most six pixels.

Regards.


自我的 Galaxy 發送



-------- 原始訊息 --------
由： olikraus ***@***.***>
日期： 23/3/2023 14:35 (GMT+08:00)
收件人： olikraus/u8g2 ***@***.***>
抄送： Ultramxn ***@***.***>, Author ***@***.***>
主題： Re: [olikraus/u8g2] Modify screen width (Discussion #2140)


What are the exact names of the constructors?

**Reply by olikraus** [reply]
In your local copy of u8g2, can you change the value 4 to 0 in the following line:

https://github.com/olikraus/u8g2/blob/5a58330f1defb65248673f6c47382c0ac5009fe8/csrc/u8x8_d_st7567.c#L1044

**Reply by Ultramxn** [reply]
Have done the change and the test screen skipping 4 more pixels on the left. However, we've found the direction to get the problem solved. The test display perfectly when offset by 10. Thank you very much !

**Reply by olikraus** [reply]
ok, so you have modified the os12864 constructor, right? It means that the constructor itself still doesn't fit to your screen size, correct?

**Reply by Ultramxn** [reply]
By changing Line 1044 as you said, the value from 4 to 10 (instead of 0 because smaller value skipping more pixels on the left of the screen), the screen test good without missing pixel horizontally (vertically has only 32, of course, being a 12232 LCD).

Best regards.

**Reply by olikraus** [reply]
Thanks for the feedback. So we keep everything as it is?

**Reply by Ultramxn** [reply]
I don't know whether it is worth to change or add new constructor for a custom lcd, however, it may leading problem to reappear after every update of the library. Anyway, many thanks to your help and I can print to the lcd without missing pixels now.
By the way, can we pass offset as a optional parameter along with the pin numbers to the constructor module? It may reduce lots of modification hereafter.

**Reply by olikraus** [reply]
> By the way, can we pass offset as a optional parameter along with the pin numbers to the constructor module? It may reduce lots of modification hereafter.

This would be a much bigger effort than just creating a new constructor... So if you are willing to test and give some feedback to a new constructor, then I will create the same.

**Reply by Ultramxn** [reply]
> So if you are willing to test and give some feedback to a new constructor, then I will create the same.

Sure, at your deposal👍

**Reply by Ultramxn** [reply]
So, did the information enough for the new constructor?

thx again.

**Reply by olikraus** [reply]
ok then, let's create an issue first...

**Comment by olikraus** [comment]
issue reference:
https://github.com/olikraus/u8g2/issues/2152
